### PR TITLE
Prepare for 8.0.0 release

### PR DESCRIPTION
**Pull request by chwarr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,182 +11,212 @@ tag versions. The Bond compiler (`gbc`) and
...
-* C++ codegen now generates lazily constructed enum name-to-value and value-to-name maps.
-  Additionally, a user-defined map type can now be provided to `GetNameToValueMap` and
-  `GetValueToNameMap`.
+* C++ codegen now properly generates move assignment operators. Previously,
+  this was broken for some cases.
+* C++ codegen no longer generates checks for C++11, except for MSVC 2013
+  workarounds.
+* C++ codegen no longer generates data member initialization that invokes a
+  constructor accepting `Comparer` for associative containers.
+* C++ codegen can now cenerate copy and move constructors with an allocator
```
Comment (lalo) [inline CHANGELOG.md]: typo: generate

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,182 +11,212 @@ tag versions. The Bond compiler (`gbc`) and
...
   Core projects.
-* The .NET Standard assemblies are fully strong-name signed. Previously,
-  they were inadvertently only public strong-name signed.
-* Fixed a bug in the codegen targets when using `gbc` from $PATH on macOS
-  and Linux that prevented the C# compiler from finding the generated C#
-  files.
-* *Preview*: Added preliminary support for generating types with
-  constructors with parameters for each field. This functionality will
-  change in the future and may be removed. [Pull request
+* The .NET Standard assemblies are fully strong-name signed. Previously, they
```
Comment (lalo) [inline CHANGELOG.md]: Do we need to mention the signature type change? sha1->sha2

Comment (chwarr) [reply CHANGELOG.md]: We can, sure.

File: CHANGELOG.md (hunk 1 above)
Comment (chwarr) [reply CHANGELOG.md]: ✔️

**Pull request merged**